library/Vanilla/DateFilterSchema.php

```diff
@@ -181,26 +221,88 @@ public function validate($data, $sparse = false) {
+        } elseif (preg_match("/^(?!$escapedOpen)$simpleOperators?(?<date>.+)/", $value, $matches)) {
+            $result = $this->parseSimple($matches['date'], $matches['op'], $field);
+        } else {
+            $field->addError('invalid', ['messageCode' => '{field} is not formatted as a valid date filter.']);
+        }
+
+        return $result;
+    }
+
+    /**
+     * If the parameter value is a valid date filter value, return an array of query conditions.
+     *
+     * @throws ServerException
+     * @param string $field The name of the field in the filters.
+     * @param mixed $dateData The decoded date data.
+     * @return array
+     */
+    public static function dateFilterField($field, array $dateData) {
+        $validOperators = ['=', '>', '<', '>=', '<=', '[]', '()', '[)', '(]'];
+        $result = [];
+
+        if (array_key_exists('operator', $dateData) && array_key_exists('date', $dateData) && is_array($dateData['date'])) {
+            $op = $dateData['operator'];
+            $dates = $dateData['date'];
+
+            if (in_array($op, $validOperators)) {
+                switch ($op) {
+                    case '>':
+                    case '<':
+                    case '>=':
+                    case '<=':
+                        if ($dates[0] instanceof DateTimeImmutable) {
+                            $result = ["{$field} {$op}" => $dates[0]];
+                        }
+                        break;
+                    case '[]':
+                    case '()':
+                    case '[)':
+                    case '(]':
+                        // DateFilterSchema has already taken care of any inclusive/exclusive range adjustments
+                        // so we can always use >= and <=
+                        $result = [
+                            "{$field} >=" => $dates[0],
+                            "{$field} <=" => $dates[1],
+                        ];
+                        break;
+                    case '=':
+                        if (count($dates) === 1) {
+                            $result = ["{$field}" => $dates[0]];
+                        } else {
+                            $result = [
+                                "{$field} >=" => $dates[0],
+                                "{$field} <=" => $dates[1],
+                            ];
+                        }
+                        break;
                 }
-            } elseif (preg_match('/^(?<op><=|>=|>|<|)?\s*(?<value>'.$this->dateRegEx.')/i', $value, $match)) {
-                $result = $this->parseSimple($match['value'], $match['op'], $field);
-            } else {
-                $field->addError('invalid', ['messageCode' => '{field} is not formatted as a valid date filter.']);
             }
         } else {
-            $field->addTypeError('string');
+            throw new Exception('Invalid data supplied to DateFilterSchema');
```

Comment:
This exception message doesn't make a lot of sense to me. Should it read `dateFilterField` instead of `DateFilterSchema`?